adapters/gst/gst_plugins/python/ffmpeg_src.py

```diff
@@ -0,0 +1,246 @@
+        elif prop.name == 'decode':
+            self._decode = value
+        elif prop.name == 'loglevel':
+            self._loglevel = value
+        else:
+            raise AttributeError(f'Unknown property {prop.name}')
+
+    def do_start(self):
+        """Gst plugin start function."""
+
+        try:
+            required_property(self._uri, 'uri')
+            self.logger.info('Creating FFMpegSource.')
+            self._ffmpeg_source = FFMpegSource(
+                self._uri,
+                params=self._params,
+                queue_len=self._queue_len,
+                decode=self._decode,
+                ffmpeg_log_level=STR_TO_FFMPEG_LOG_LEVEL[self._loglevel.lower()],
+            )
+
+        except Exception as exc:
+            self.logger.exception('Failed to start element: %s.', exc, exc_info=True)
+            frame = inspect.currentframe()
+            propagate_gst_setting_error(self, frame, __file__, text=exc.args[0])
+            return False
+
+        return True
+
+    def do_create(self, offset: int, size: int, buffer: Gst.Buffer = None):
+        """Create gst buffer."""
+
+        self.logger.debug('Receiving next frame')
+
+        frame: VideoFrameEnvelope = self._ffmpeg_source.video_frame()
+        self.logger.debug(
+            'Received frame with codec %s, PTS %s and DTS %s.',
+            frame.codec,
+            frame.pts,
+            frame.dts,
+        )
+        self.logger.debug(
+            '%s frames in queue, %s frames skipped.',
+            frame.queue_len,
+            frame.queue_full_skipped_count,
+        )
+        if frame.queue_full_skipped_count > self._last_skipped_frames:
+            self.logger.warning(
+                'Skipped %s frames due to queue overflow.',
+                frame.queue_full_skipped_count - self._last_skipped_frames,
+            )
+            self._last_skipped_frames = frame.queue_full_skipped_count
+        frame_params = FrameParams(
+            codec_name=frame.codec,
+            width=frame.frame_width,
+            height=frame.frame_width,
+            framerate=frame.fps,
+        )
+        if self._frame_params != frame_params:
+            self.on_frame_params_change(frame_params)
+        buffer: Gst.Buffer = Gst.Buffer.new_wrapped(frame.payload_as_bytes())
+        tb_num, tb_denum = frame.time_base
+        buffer.pts = frame.pts * tb_num * Gst.SECOND // tb_denum
+        buffer.dts = frame.dts * tb_num * Gst.SECOND // tb_denum
+
+        self.logger.debug(
+            'Pushing buffer of size %s with PTS=%s, DTS=%s and duration=%s to src pad.',
+            buffer.get_size(),
+            buffer.pts,
+            buffer.dts,
+            buffer.duration,
+        )
+
+        return Gst.FlowReturn.OK, buffer
+
+    def on_frame_params_change(self, frame_params: FrameParams):
+        """Change caps when video parameter changed."""
+
+        self._frame_params = frame_params
+        codec = CONVERT_CODEC[frame_params.codec_name]
```

Comment:
Please add a check and termination with a correct message that codec is not supported if not in list. The current termination is not user-friendly.